File: Chapter_two/chapt2_ws/src/demo_cpp_pkg/src/person_node.cpp
#include "rclcpp/rclcpp.hpp"

class PersonNode : public rclcpp::Node
{
private:
    std::string name_values;
    int age_values;

public:
    PersonNode(const std::string &node_name, const std::string &name, const int &age)
        /*调用父类的构造函数，等同于python中的 super().__init__()*/
        : Node(node_name)
    {
        this->name_values = name;
        this->age_values = age;
    };

    void eat(const std::string &food_name)
    {
        RCLCPP_WARN(this->get_logger(), "%d岁的%s正在吃%s", this->age_values, this->name_values.c_str(), food_name.c_str());
    }
};

int main(int argc, char *argv[])
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<PersonNode>("person_node", "zhangsan", 18);
    RCLCPP_INFO(node->get_logger(), "节点已经创建成功了");
    node->eat("鱼香ROS");
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}

File: Chapter_two/chapt2_ws/src/demo_python_pkg/demo_python_pkg/person_node.py
import rclpy 
from rclpy.node import Node

class PersonNode(Node):
    def __init__(self, node_name: str, name_value: str, age_value: int) -> None:
        print('PersonNode __init__ 方法被调用了，添加了两个属性')
        super().__init__(node_name)
        self.name = name_value
        self.age = age_value
    
    def eat(self, food_name: str):
        """
        方法：吃东西
        :food_name: 食物名称
        """
        # print(f"{self.age}岁的{self.name}，爱吃{food_name}")
        self.get_logger().info(f"{self.age}岁的{self.name}，爱吃{food_name}")

def main():
    rclpy.init()
    node = PersonNode('zhangsan', '法外狂徒张三', 18)
    node.eat("鱼香ROS")
    rclpy.spin(node)
    rclpy.shutdown()
